File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");                     //Prompt for text
    int wc = 1;                                             //Define word count
    for (int i = 0; i < strlen(text); i++)
    {
        if (text[i] == ' ' && text[i + 1] != ' ')
        {
            wc = wc + 1;
        }
    }
    float l = 0;                                            //Define number of letters
    for (int i = 0; text[i] != '\0'; i++)
    {
        if ((65 <= (int)text[i] && (int)text[i] <= 90) || (97 <= (int)text[i] && (int)text[i] <= 122))
        {
            l = l + 1;
        }
    }
    float L = l * 100 / wc;                                 //Calculate number of letters per 100 wc

    float s = 0;                                            //Define number of sentences
    for (int i = 0; text[i] != '\0'; i++)
    {
        if ((int)text[i] == 33 || (int)text[i] == 46 || (int)text[i] == 63)
        {
            s = s + 1;
        }
    }
    float S = s * 100 / wc;                                 //Calculate number of sentences per 100 wc
    float index = 0.0588 * L - 0.296 * S - 15.8;            //Calculate Coleman-Liau index

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    printf("Grade %.0f\n", index);

}

File: readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    # text.split separates words in text- len counts those words
    wc = len(text.split())
    
    l = 0
    # for counting letters, count when letter fall within lowercase and uppercase alphabet range
    for i in range(0, len(text)):
        if (64 < ord(text[i]) and ord(text[i]) < 91) or (96 < ord(text[i]) and ord(text[i]) < 123):
            l = l + 1
    L = (l / wc) * 100
    
    s = 0
    # for counting sentences, count when there is . ! or ?
    for i in range(0, len(text)): 
        if ord(text[i]) == 33 or ord(text[i]) == 46 or ord(text[i]) == 63:
            s = s + 1
    S = (s / wc) * 100
    
    # Coleman-Liau index
    C = 0.0588 * L - 0.296 * S - 15.8
    
    if C < 1:
        print("Before Grade 1")
    elif C >= 1 and C <= 16:
        print(f"Grade {round(C)}")
    else:
        print("Grade 16+")
    
        
main()